internal/compiler/expand.go

```diff
@@ -149,6 +150,9 @@ func (c *Compiler) expandStmt(qc *QueryCatalog, raw *ast.RawStmt, node ast.Node)
 				if counts[cname] > 1 {
 					cname = tableName + "." + cname
 				}
+				if _, ok := c.parser.(*sqlite.Parser); ok && column.DataType == "jsonb" {
```

Comment:
So I left the `*sqlite.Parser` in here as a dummy implementation for detecting when we're running as SQLite, but I could use some help determining what the preferred convention should be. It feels a little heavy handed to make it part of one of the interfaces like `Parser` given it's such an unusual corner case.

Reply:
I looked through the code to see if there was a better place to put this. The ideal solution is to create a `SelectExpr` function on the compiler.Column which adds the `json()`. That said, we'd have to pipe it through the catalog and the ast, all of which are engine-agnostic.

Instead, let's create a new `Selector` interface with a single method `Expr(name, column) string`. In NewCompiler we already switch on the engine so we can pick the correct one to use. Unlike the parser and the catalog, the selector structs shouldn't be exported and should live in the compiler package.

What do you think?